x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -310,6 +310,55 @@ func (m *MongoCrypt) createExplicitEncryptionContext(opts *options.ExplicitEncry
 		}
 	}
 
+	if opts.TextOptions != nil {
+		idx, mongocryptDoc := bsoncore.AppendDocumentStart(nil)
+		if opts.TextOptions.Substring != nil {
+			substringIdx, substringDoc := bsoncore.AppendDocumentStart(nil)
+			substringDoc = bsoncore.AppendInt32Element(substringDoc, "strMaxLength", opts.TextOptions.Substring.StrMaxLength)
+			substringDoc = bsoncore.AppendInt32Element(substringDoc, "strMinQueryLength", opts.TextOptions.Substring.StrMinQueryLength)
+			substringDoc = bsoncore.AppendInt32Element(substringDoc, "strMaxQueryLength", opts.TextOptions.Substring.StrMaxQueryLength)
+			substringDoc, err := bsoncore.AppendDocumentEnd(substringDoc, substringIdx)
+			if err != nil {
+				return nil, err
```

Comment:
Optional: Add context to the error.

```suggestion
				return nil, fmt.Errorf("error building substring doc: %w", err)
```